### ISSUE TITLE
last member in dynamic array is not checked

### ISSUE THREAD

**Issue by tserg** [vyperlang/vyper#2648]
### Version Information

* vyper Version (output of `vyper --version`): 0.3.2+commit.3b0d3cf9.dirty
* OS: linux
* Python Version (output of `python --version`): 3.8.5

### What's your issue about?

Last member of dynamic array is not checked:
```
@external
def check(a: uint256) -> bool:
    x: DynArray[uint256, 2] = [3, 7]
    return a in x
```
Running `check(7)` returns False. If I change the array to `[3, 7, 4]`, `check(7)` now returns True but `check(4)` returns False.

### How can it be fixed?

Fill this in if you know how to fix it.

**Comment by tserg** [comment]
This issue does not arise for static arrays.

**Comment by charles-cooper** [comment]
Seems to have been fixed in #2649; double check before release.

**Comment by charles-cooper** [comment]
verified with titanoboa(bd77b21fabc38b665fd02e0ba1ddb1d148546d76) that as of 09f37635, the issue is fixed

```
In [1]: %load_ext boa.ipython

In [2]: %%vyper
   ...: @external
   ...: def check(a: uint256) -> bool:
   ...:     x: DynArray[uint256, 3] = [3, 7, 4]
   ...:     return a in x
   ...: 
Out[2]: <boa.contract.VyperDeployer at 0x7f4a5e1c02b0>

In [3]: c = _.deploy()

In [4]: str(c)
Out[4]: '<VyperContract at 0x0000000000000000000000000000000000000066, compiled with vyper-0.3.4+09f37635>'

In [5]: c.check(7), c.check(4)
Out[5]: (True, True)
```
